### PR TITLE
Update the IOU "Cancel" button 

### PR DESCRIPTION
**Pull request by thesahindia**
### Details

- Changed the way we show ActivityIndicator inside button. If isLoading is true the text for button will be rendered behind the ActivityIndicator so that button width doesn't change when it's in loading state.

- Size of iou cancel/decline button changes while loading, this PR fixes that. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/8715

### Tests | QA Steps

1. Go to a chat and request money
2. Check the iou details and cancel the request
3. Verify the button width doesn't change
4. Follow the same steps above but with language set to spanish

---
1. Go to a chat and request money from account b
2. Now login to account b
3. Check the iou request
4. Verify the button width doesn't change
5. Follow the same steps above but with language set to spanish

1. Sign out
2. Enter email or phone that doesn't belong to an existing account
3. Press "Continue"
4. Tap on resend link button
5. Verify the button width doesn't change
6. Do the same again but with language set to spanish

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

#### Web

https://user-images.githubusercontent.com/83179295/169117495-b8731c67-183a-4b8b-a109-d6a69a5a52d2.mov

#### Mobile Web

https://user-images.githubusercontent.com/83179295/169117483-29e8ba4c-2c8f-4a86-b7f8-df2cc589d847.mov

#### Desktop

https://user-images.githubusercontent.com/83179295/169117985-8b69cbd4-8171-47ef-9d3d-ca60241fdfb6.mov

#### iOS

https://user-images.githubusercontent.com/83179295/169117490-2468d740-a75a-463f-9a1e-c437ff5a127e.mov

#### Android

https://user-images.githubusercontent.com/83179295/169117453-1705259a-c506-4513-930e-108fce689ddc.mov

### CODE REVIEW

**Review by roryabraham — DISMISSED** [review]
LGTM, but I'll let @Santhosh-Sellavel test and review 👍

**Comment by Santhosh-Sellavel** [comment]
Just waiting for this thread [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1651695607990559) to conclude.

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
We are looking for another approach as discussed! https://expensify.slack.com/archives/C01GTK53T8Q/p1652390743738389?thread_ts=1651695607.990559&cid=C01GTK53T8Q

File: src/components/Button.js [hunk 1]
```diff
@@ -173,14 +173,11 @@ class Button extends Component {
...
 
-        if (this.props.isLoading) {
-            return <ActivityIndicator color={themeColors.textReversed} />;
-        }
-
         const textComponent = (
             <Text
                 selectable={false}
                 style={[
+                    this.props.isLoading && styles.opacity0,
```
Comment (thesahindia) [inline src/components/Button.js:180]: using opacity 0 as visibility hidden doesn't work on native

File: src/pages/signin/ResendValidationForm.js [hunk 2]
```diff
@@ -143,7 +143,6 @@ class ResendValidationForm extends React.Component {
                         text={this.props.translate('resendValidationForm.resendLink')}
                         isLoading={this.props.account.loading}
                         onPress={this.validateAndSubmitForm}
-                        style={styles.resendLinkButton}
```
Comment (thesahindia) [inline src/pages/signin/ResendValidationForm.js:146]: removed the minWidth given here as we don't need this one now.

**Comment by thesahindia** [comment]
Just after pressing the "resend link" button we are showing "Link has been re-sent" but it's kinda weird that the resend link button keeps loading for a while. 

What you think @roryabraham 

https://user-images.githubusercontent.com/83179295/169136567-69a55ad4-9a11-4e12-b560-263f849c5d90.mov

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

File: src/components/Button.js (hunk 1 above)
Comment (thesahindia) [reply src/components/Button.js:180]: I can't understand the question here. Can you rephrase it?

Comment (Santhosh-Sellavel) [reply src/components/Button.js:180]: > using opacity 0 as visibility hidden doesn't work on native

What have you done instead here to solve the problem?

Comment (thesahindia) [reply src/components/Button.js:180]: I think you misunderstood the comment, let me rephrase it for you. "Visibility hidden doesn't work on native so instead of it I am using opacity 0"

Comment (Santhosh-Sellavel) [reply src/components/Button.js:180]: There's a lot rephrases here 😄

**Comment by thesahindia** [comment]
bump @Santhosh-Sellavel, the review is still pending.

**Comment by Santhosh-Sellavel** [comment]
I will get this soon, @thesahindia can you resolve the conflicts!

File: src/styles/styles.js [hunk 3]
```diff
@@ -2520,6 +2520,10 @@ const styles = {
         marginBottom: 40,
         padding: 16,
     },
+
+    iouRejectButton: {
+        minWidth: 69,
+    },
```
Comment (Santhosh-Sellavel) [inline src/styles/styles.js:2526]: Remove this

**Comment by thesahindia** [comment]
Done!

File: src/components/Button.js [hunk 4]
```diff
@@ -273,6 +270,12 @@ class Button extends Component {
                         ]}
                     >
                         {this.renderContent()}
+                        {this.props.isLoading && (
+                        <ActivityIndicator
```
Comment (roryabraham) [inline src/components/Button.js:274]: This should be indented 4 more spaces.

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
Sorry, meant to hit `Request Changes`, not `Approve`

**Review by roryabraham — CHANGES_REQUESTED** [review]
Approved by mistake

**Comment by roryabraham** [comment]
> Loading Indicator is not prominent as before, looks dull!

Agree, we should switch the `color` of the `ActivityIndicator` from `themeColors.textReversed` to `themeColors.text`

**Comment by thesahindia** [comment]
> Agree, we should switch the `color` of the `ActivityIndicator` from `themeColors.textReversed` to `themeColors.text`

We can add a new prop for passing a different color and and we can keep `themeColors.textReversed` as default because `themeColors.text` won't look good on green buttons.

**Comment by Santhosh-Sellavel** [comment]
> > Loading Indicator is not prominent as before, looks dull!
> 
> Agree, we should switch the `color` of the `ActivityIndicator` from `themeColors.textReversed` to `themeColors.text`

but it would affect other buttons
Ex: Save button

**Comment by roryabraham** [comment]
Ah, good point @thesahindia. Rather than adding another prop, can we just do something like:

```jsx
<ActivityIndicator color={(this.props.success || this.props.danger) ? themeColors.textReversed : themeColors.text} />
```

?

**Comment by thesahindia** [comment]
can we just do something like:
> 
> ```js
> <ActivityIndicator color={(this.props.success || this.props.danger) ? themeColors.textReversed : themeColors.text} />
> ```

Oh yeah I didn't think about this, this is better 👍🏼

**Comment by Santhosh-Sellavel** [comment]
<h4> #### PR Reviewer Checklist</h4>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

**Review by Santhosh-Sellavel — APPROVED** [review]
Looks good, and tests well for me!

**Comment by Santhosh-Sellavel** [comment]
@roryabraham all you!

**Review by roryabraham — APPROVED** [review]
Thanks, LGTM 👍🏼
